### PR TITLE
Bug 1498912, updated container technology versions and added support statements

### PR DESCRIPTION
**Pull request by ahardin-rh**
This addresses https://bugzilla.redhat.com/show_bug.cgi?id=1498912 and also https://bugzilla.redhat.com/show_bug.cgi?id=1476927 (regarding support statements).

@wgordon17 @luciddreamz @adellape @sferich888 PTAL

### CODE REVIEW

File: dev_guide/migrating_applications/web_framework_applications.adoc [hunk 1]
```diff
@@ -48,17 +48,20 @@ python:3.3~https://github.com/<github-id>/<repo-name> --name=<app-name> -e
...
 [[dev-guide-migrating-web-framework-applications-supported-python-versions]]
 *Supported Python Versions*
-
+ifndef::openshift-online[]
 [cols="2,2",options="header"]
 |===
 |v2 |v3
 
-|Python: 2.6, 2.7, 3.3   |Python: 2.7, 3.3, 3.4
+|Python: 2.6, 2.7, 3.3   |link:https://www.openshift.com/features/containers.html[Supported Container Images]
```
Comment (sferich888) [inline dev_guide/migrating_applications/web_framework_applications.adoc]: I don't think this points to the right content (set of support images).

File: dev_guide/migrating_applications/web_framework_applications.adoc [hunk 2]
```diff
@@ -48,17 +48,20 @@ python:3.3~https://github.com/<github-id>/<repo-name> --name=<app-name> -e
...
 
 |Django     |Django-psql-example (quickstart)
 
 |===
-
+endif::[]
+ifdef::openshift-online[]
+See link:https://www.openshift.com/features/containers.html[Supported Container Images].
```
Comment (sferich888) [inline dev_guide/migrating_applications/web_framework_applications.adoc]: I don't think this points to the right content (set of support images).

File: dev_guide/migrating_applications/web_framework_applications.adoc [hunk 3]
```diff
@@ -100,19 +103,22 @@ ruby:2.0~https://github.com/<github-id>/<repo-name>.git
...
 [[dev-guide-migrating-web-framework-applications-supported-ruby-versions]]
 *Supported Ruby Versions*
-
+ifndef::openshift-online[]
 [cols="2,2",options="header"]
 |===
 |v2 |v3
 
-|Ruby: 1.8, 1.9, 2.0   |Ruby: 2.0, 2.2
+|Ruby: 1.8, 1.9, 2.0   |link:https://www.openshift.com/features/containers.html[Supported Container Images]
```
Comment (sferich888) [inline dev_guide/migrating_applications/web_framework_applications.adoc]: I don't think this points to the right content (set of support images).

File: dev_guide/migrating_applications/web_framework_applications.adoc [hunk 4]
```diff
@@ -100,19 +103,22 @@ ruby:2.0~https://github.com/<github-id>/<repo-name>.git
...
 
 |Ruby on Rails: 3, 4     |Rails-postgresql-example (quickstart)
 
 |Sinatra    |
 
 |===
-
+endif::[]
+ifdef::openshift-online[]
+See link:https://www.openshift.com/features/containers.html[Supported Container Images].
```
Comment (sferich888) [inline dev_guide/migrating_applications/web_framework_applications.adoc]: I don't think this points to the right content (set of support images).

File: dev_guide/migrating_applications/web_framework_applications.adoc [hunk 5]
```diff
@@ -139,12 +145,12 @@ $ oc new-app https://github.com/<github-id>/<repo-name>.git
...
 [[dev-guide-migrating-web-framework-applications-supported-PHP-versions]]
 *Supported PHP Versions*
-
+ifndef::openshift-online[]
 [cols="2,2",options="header"]
 |===
 |v2 |v3
 
-|PHP: 5.3, 5.4   |PHP:5.5, 5.6
+|PHP: 5.3, 5.4   |link:https://www.openshift.com/features/containers.html[Supported Container Images]
```
Comment (sferich888) [inline dev_guide/migrating_applications/web_framework_applications.adoc]: I don't think this points to the right content (set of support images).

File: dev_guide/migrating_applications/web_framework_applications.adoc [hunk 6]
```diff
@@ -157,7 +163,10 @@ $ oc new-app https://github.com/<github-id>/<repo-name>.git
 | |cakephp-mysql-example (quickstart)
 
 |===
-
+endif::[]
+ifdef::openshift-online[]
+See link:https://www.openshift.com/features/containers.html[Supported Container Images].
```
Comment (sferich888) [inline dev_guide/migrating_applications/web_framework_applications.adoc]: I don't think this points to the right content (set of support images).

File: dev_guide/migrating_applications/web_framework_applications.adoc [hunk 7]
```diff
@@ -214,17 +223,20 @@ $ oc new-app https://github.com/<github-id>/<repo-name>.git
...
 [[dev-guide-migrating-web-framework-applications-supported-perl-versions]]
 *Supported Perl Versions*
-
+ifndef::openshift-online[]
 [cols="2,2",options="header"]
 |===
 |v2 |v3
 
-|Perl: 5.10  |Perl: 5.16, 5.20
+|Perl: 5.10  |link:https://www.openshift.com/features/containers.html[Supported Container Images]
```
Comment (sferich888) [inline dev_guide/migrating_applications/web_framework_applications.adoc]: I don't think this points to the right content (set of support images).

File: dev_guide/migrating_applications/web_framework_applications.adoc [hunk 8]
```diff
@@ -214,17 +223,20 @@ $ oc new-app https://github.com/<github-id>/<repo-name>.git
...
 
 |    |Dancer-mysql-example (quickstart)
 
 |===
-
+endif::[]
+ifdef::openshift-online[]
+See link:https://www.openshift.com/features/containers.html[Supported Container Images].
```
Comment (sferich888) [inline dev_guide/migrating_applications/web_framework_applications.adoc]: I don't think this points to the right content (set of support images).

File: dev_guide/migrating_applications/web_framework_applications.adoc [hunk 9]
```diff
@@ -271,17 +283,20 @@ $ oc new-app https://github.com/<github-id>/<repo-name>.git
...
 [[dev-guide-migrating-web-framework-applications-supported-Node.js-versions]]
 *Supported Node.js Versions*
-
+ifndef::openshift-online[]
 [cols="2,2",options="header"]
 |===
 |v2 |v3
 
-|Node.js 0.10  |Nodejs: 0.10, 4, and 6
+|Node.js 0.10  |link:https://www.openshift.com/features/containers.html[Supported Container Images]
```
Comment (sferich888) [inline dev_guide/migrating_applications/web_framework_applications.adoc]: I don't think this points to the right content (set of support images).

**Review by sferich888 — COMMENTED** [review]
We need to have a discussion about where we document the supported image, and what it means to be a supported image. 

See https://access.redhat.com/articles/2176281

**Comment by ahardin-rh** [comment]
@luciddreamz Thoughts on the above comments?

**Comment by luciddreamz** [comment]
@ahardin-rh Seems fine. 

For reference:

- https://www.openshift.com/features/containers.html#online3
- https://www.openshift.com/features/containers.html#dedicated
- https://www.openshift.com/features/containers.html#enterprise30
- https://www.openshift.com/features/containers.html#enterprise31
- https://www.openshift.com/features/containers.html#enterprise32
- ...
- https://www.openshift.com/features/containers.html#enterprise36

**Comment by wgordon17** [comment]
@luciddreamz I didn't know about those anchor links!

@ahardin-rh It may be beneficial (although painful) to use the specific anchor links for each docs version.

@sferich888 The content on https://www.openshift.com/features/containers.html comes from the official image streams included in the ansible installers. Should this not be considered the "source of truth" for supported containers?

**Comment by ahardin-rh** [comment]
@luciddreamz oh nice! Thank you. That's very helpful. 

@wgordon17 Yes, I will create separate PRs to direct users to the specific anchors now that I know those are available. That makes these updates much more usable. 👍

**Comment by ahardin-rh** [comment]
Updated with the appropriate anchors and ifdef statements. I will create a separate PR for 3.4 and another for 3.5. Before I do that, @wgordon17 PTAL at this updated PR and let me know your thoughts. Thanks!

File: dev_guide/migrating_applications/web_framework_applications.adoc [hunk 10]
```diff
@@ -271,33 +295,50 @@ $ oc new-app https://github.com/<github-id>/<repo-name>.git
...
 [[dev-guide-migrating-web-framework-applications-supported-Node.js-versions]]
 *Supported Node.js Versions*
-
+ifdef::openshift-enterprise,openshift-origin[]
 [cols="2,2",options="header"]
 |===
 |v2 |v3
 
-|Node.js 0.10  |Nodejs: 0.10, 4, and 6
+|Node.js 0.10  |link:https://www.openshift.com/features/containers.html[Supported Container Images]
```
Comment (wgordon17) [inline dev_guide/migrating_applications/web_framework_applications.adoc]: #enterprise36 anchor here?

File: dev_guide/migrating_applications/web_framework_applications.adoc [hunk 11]
```diff
@@ -271,33 +295,50 @@ $ oc new-app https://github.com/<github-id>/<repo-name>.git
...
 
+[IMPORTANT]
+====
+Currently, support for migrating WordPress applications is offered by the
+community only and not by Red Hat support.
+====
+
 For guidance on migrating WordPress applications to {product-title} v3, see the
 link:https://blog.openshift.com/migrating-wordpress-openshift-3/[OpenShift
 blog].
```
Comment (wgordon17) [inline dev_guide/migrating_applications/web_framework_applications.adoc]: So this was just posted (https://blog.openshift.com/running-wordpress-easy-way/)

@luciddreamz Thoughts on including the post here as a "for new wordpress applications see https://blog.openshift.com/running-wordpress-easy-way/"? Or leave no mention of it?

Comment (luciddreamz) [reply dev_guide/migrating_applications/web_framework_applications.adoc]: Probably don't mention it. If Graham ever publishes his new [blog article](https://docs.google.com/document/d/1lXoQ2XjdSwCXff-0DhVpDHhFFwbtl_U-fd04JajC-nM/edit) and [Quickstart](https://github.com/openshift-evangelists/wordpress-quickstart), we should link to that, but probably not to this one.

File: dev_guide/migrating_applications/web_framework_applications.adoc (hunk 10 above)
Comment (ahardin-rh) [reply dev_guide/migrating_applications/web_framework_applications.adoc]: Fixed!

**Comment by ahardin-rh** [comment]
This is updated. Thanks!

**Comment by wgordon17** [comment]
LGTM =)

**Comment by ahardin-rh** [comment]
No revision history needed

File: dev_guide/migrating_applications/web_framework_applications.adoc (hunk 11 above)
Comment (sferich888) [reply dev_guide/migrating_applications/web_framework_applications.adoc]: Should we even be claiming support for OpenShift Online? We in CEE don't directly support Wordpress (as we don't have a RH provided Wordpress image (or template), just community driven items.

Comment (wgordon17) [reply dev_guide/migrating_applications/web_framework_applications.adoc]: @sferich888 I think that's the intention of adding the note here, to call out that RH does *not* support Wordpress

Comment (ahardin-rh) [reply dev_guide/migrating_applications/web_framework_applications.adoc]: Correct. Those notes address https://bugzilla.redhat.com/show_bug.cgi?id=1476927

**Pull request merged**